mlflow/utils/search_utils.py

```diff
@@ -1910,42 +1910,116 @@ def model_matches(model):
+        return field_name, ascending, dataset_name, dataset_digest
+
+    @classmethod
+    def _apply_reversor_for_logged_model(
+        cls,
+        model: LoggedModel,
+        key: str,
+        ascending: bool,
+        dataset_name: Optional[str],
+        dataset_digest: Optional[str],
+    ):
+        if "." in key:
+            metric_key = key.split(".")[1]
+            metric_values = [
+                m.value
+                for m in model.metrics
+                if m.key == metric_key
+                and (dataset_name is None or m.dataset_name == dataset_name)
+                and (dataset_digest is None or m.dataset_digest == dataset_digest)
+            ]
+            value = None if len(metric_values) == 0 else max(metric_values)
+            return _LoggedModelMetricComp(value) if ascending else _LoggedModelMetricReversor(value)
+        else:
+            value = getattr(model, key)
+        return value if ascending else _Reversor(value)
 
     @classmethod
-    def _get_sort_key(cls, order_by_list):
-        order_by = []
-        parsed_order_by = map(cls.parse_order_by_for_logged_models, order_by_list or [])
-        for type_, key, ascending in parsed_order_by:
-            if type_ == "attribute":
-                order_by.append((key, ascending))
-            else:
-                raise MlflowException.invalid_parameter_value(f"Invalid order_by entity: {type_}")
+    def _get_sort_key(cls, order_by_list: Optional[list[dict[str, Any]]]):
+        parsed_order_by = list(map(cls.parse_order_by_for_logged_models, order_by_list or []))
 
         # Add a tie-breaker
-        if not any(key == "creation_timestamp" for key, _ in order_by):
-            order_by.append(("creation_timestamp", False))
-        if not any(key == "model_id" for key, _ in order_by):
-            order_by.append(("model_id", True))
+        if not any(key == "creation_timestamp" for key, _, _, _ in parsed_order_by):
+            parsed_order_by.append(("creation_timestamp", False, None, None))
+        if not any(key == "model_id" for key, _, _, _ in parsed_order_by):
+            parsed_order_by.append(("model_id", True, None, None))
 
         return lambda logged_model: tuple(
-            _apply_reversor(logged_model, k, asc) for (k, asc) in order_by
+            cls._apply_reversor_for_logged_model(logged_model, k, asc, dataset_name, dataset_digest)
+            for (k, asc, dataset_name, dataset_digest) in parsed_order_by
         )
 
     @classmethod
     def sort(cls, models, order_by_list):
         return sorted(models, key=cls._get_sort_key(order_by_list))
 
 
+class _LoggedModelMetricComp:
+    def __init__(self, obj):
+        self.obj = obj
+
+    def __eq__(self, other):
+        return other.obj == self.obj
+
+    def __lt__(self, other):
+        if self.obj is None:
+            return False
+        if other.obj is None:
+            return True
+        return self.obj < other.obj
+
+
+class _LoggedModelMetricReversor:
```

Comment:
Curious why we need to create this class. Can we reuse `_Reversor`?

Reply:
Emm do we allow None in other attributes field? If yes then we can reuse _Reversor

Reply:
> do we allow None in other attributes field?

Yes, there a few nullable fields in attributes.